packages/ts-interface-generator/README.md

```diff
@@ -0,0 +1,84 @@
+# @ui5/ts-interface-generator
+
+Enables UI5 control development in TypeScript by generating TypeScript interfaces which declare the methods generated by UI5 at runtime.
+
+<b>IMPORTANT: this tool is ALPHA-quality work in progress right now, with plenty of limitations! And there are changes ahead which will likely force users to adapt. Use at your own risk to give feedback. As long as it doesn't crash along the way, it should already be capable of creating nice control interfaces which will save you quite some manual work.<br>
+Note the "TODO" section below as well.<br>
+Also note that this tool does not transpile the TypeScript controls to JavaScript! It only ensures TypeScript development can happen smoothly without being blocked by TypeScript not knowing the methods and structures generated by UI5 at runtime.</b>
+
+## Usage
+
+Go into a directory containing a TypeScript project which includes UI5 controls written in TypeScript (written as ES6 classes!). When you do it for the first time in a project, install this generator as dev dependency:
+
+```sh
+npm install --save-dev @ui5/ts-interface-generator
+```
+
+_NOTE: this will only work once the first npm release has been done!_
+
+Then, to start the generator, run:
+
+```sh
+npx @ui5/ts-interface-generator --watch
```

Comment:
Interesting is one detail here Shachar made me thinking of: `npx` also installs the missing dependencies. But in general, I prefer to also install the dependency as dev dependency since `npx` first checks the local .bin folder and then falls back to the global one. 

Interesting fact: https://www.npmjs.com/package/npx

> Executes `<command>` either from a local `node_modules/.bin`, or from a central cache, installing any packages needed in order for `<command>` to run.